packages/backend-core/src/redis/redlockImpl.ts

```diff
@@ -105,13 +106,39 @@ export async function doWithLock<T>(
   task: () => Promise<T>
 ): Promise<RedlockExecution<T>> {
   const redlock = await getClient(opts.type, opts.customOptions)
-  let lock
+  let lock: Redlock.Lock | undefined
+  let timeout
   try {
     const name = getLockName(opts)
 
     // create the lock
     lock = await redlock.lock(name, opts.ttl)
 
+    if (opts.type === LockType.AUTO_EXTEND) {
+      // We keep extending the lock while the task is running
+      const extendInIntervals = (): void => {
+        timeout = setTimeout(async () => {
+          let isExpired = false
+          try {
+            lock = await lock!.extend(opts.ttl)
+          } catch (err: any) {
+            isExpired = err.message.includes("Cannot extend lock on resource")
+            if (isExpired) {
+              console.error("The lock expired", { name })
```

Comment:
Do we need to throw an error of some sort here? Theres a bit of a risk of undefined behaviour, the process is still running but the lock has timed out/been lost - another process could start running on the resources now and then both break.

Reply:
True, remove the try/catch. If there is an error because the lock expired, the lock itself should throw an exception and kill the tread. I will check if I can write a test for it

Reply:
> I was reading https://brooker.co.za/blog/2023/11/27/about-time.html recently and it mentions a "safety period" on locks that lasts after the expiry where the lock cannot be given out again, in order to handle clock drift and so on. Is that something we have / can configure in Redis?

Redlock by default uses a `0.01` factor on the TTL to adjust for drift during locks/unlocking, I believe that is automatically factored into the TTL release time, instead of deleting the entry it sets the final expiry based on the TTL + drift factor.